### ISSUE TITLE
Farming for Blockheads merchant blood bug

### ISSUE THREAD

**Issue by egodsk** [Darkosto/SevTech-Ages#1487]
## Issue / Bug
Merchants create blood in the Tinkers Construct, which can be used to create unlimited blood, because each time the merchant table is placed a new merchant is spawned.

## Current Behavior
Merchants create blood in smeltery when hurt. Should not happen.

## Possible Solution
The way that Tinkers Construct adds blood when an entity is hurt is done with the following code:
```java
// custom melting?
FluidStack fluid = TinkerRegistry.getMeltingForEntity(entity);
// no custom melting but a living entity that's alive?
if(fluid == null && entity instanceof EntityLivingBase) {
  if(entity.isEntityAlive() && !entity.isDead) {
    fluid = new FluidStack(TinkerFluids.blood, 20);
  }
}

if(fluid != null) {
  // hurt it
  if(entity.attackEntityFrom(smelteryDamage, 2f)) {
    // spill the blood
    liquids.fill(fluid.copy(), true);
  }
}
```
Which basically means that if there is no custom melting for that entity it will check if it is alive and then add the blood. However if there was a custom melting for that entity it would skip the part of adding blood.

Therefore my suggestions comes by adding via ModTweaker a script that basically does something like this:
```java
....addEntityMelting(farmingforblockheads:merchant,"dummy entity that melts to nothing - maybe air?")
```
Which basically adds a "dummy" melting entity to the merchant entity, which therefore MAY just try and smelt the "dummy" entity (could be air?), which melts to nothing and therefore never adds the blood. 

DISCLAIMER! I do not know if this will work or not. Just a thought i had, which might fix the bug of getting infinite blood.

## Steps to Reproduce (for bugs)

1. Create a Tinkers Construct smeltery
2. Place a Merchant table so the merchant falls in the smeltery
3. The merchant will lose health and add blood to the smeltery.
4. Destroy the merchant table and place it again and a new merchant will spawn
4. Repeat for infinite blood.

## Context
Just trying to help make this great modpack better 👍

**Comment by justinrusso** [comment]
Thanks for taking the time to investigate this further

Note to self:
http://crafttweaker.readthedocs.io/en/latest/#Mods/Modtweaker/TConstruct/Melting/
**You can just override an existing entity melting recipe, you don't need to remove it first!**
```
Melting.addEntityMelting(<entity:minecraft:villager>, <liquid:water> * 0);
Melting.addEntityMelting(<entity:farmingforblockheads:merchant>, <liquid:water> * 0);
```

**Comment by justinrusso** [comment]
For the moment this wont be an option. I'm getting a NPE when trying to reference the merchant entity.

See more here: https://github.com/blay09/FarmingForBlockheads/issues/28

**Comment by egodsk** [comment]
Farming for Blockheads has officially been updated, which should include a fix for the entity loading:

https://minecraft.curseforge.com/projects/farming-for-blockheads/files/2549222

The issue on their github has been closed:
https://github.com/blay09/FarmingForBlockheads/issues/28

Great work Justin!

**Comment by egodsk** [comment]
“* Remove Villagers melting into Emerald in Tinker's smeltery
* Disabled the exploit of using the Farming for Blockheads Merchant as a way to collect Life Essence in Blood Magic”

Has this been fixed? (Sene in changelog for 3.0.5) (Asking in case we can close the issue)

**Comment by justinrusso** [comment]
No, none of that is about the tcon melting into blood.
